pkg/agent/core/ngt/service/ngt.go

```diff
@@ -389,6 +470,136 @@ func (n *ngt) load(ctx context.Context, path string, opts ...core.Option) (err e
 	return nil
 }
 
+// backupBroken backup index at originPath into brokenDir.
+// The name of the directory will be timestamp(UnixNano).
+// If it exeeds the limit, backupBroken removes the oldest backup directory.
+func backupBroken(originPath string, brokenDir string, limit int) error {
+	if limit <= 0 {
+		return nil
+	}
+
+	// do nothing when origin path is empty
+	files, err := file.ListInDir(originPath)
+	if err != nil {
+		return err
+	}
+	if len(files) == 0 {
+		return nil
+	}
+
+	// how many generation exists in the path?
+	files, err = file.ListInDir(brokenDir)
+	if err != nil {
+		return err
+	}
+
+	if len(files) >= limit {
+		// remove the oldest
+		log.Infof("There's already more than %v broken index generations stored. Thus removing the oldest.", limit)
+		sort.Slice(files, func(i, j int) bool {
```

Comment:
you can use internal/slices package for sorting.

Reply:
[c55b051](https://github.com/vdaas/vald/pull/2034/commits/c55b05189c34b7d6c7174ec55b5a693f353d6201)